tokenizers/Cargo.toml

```diff
@@ -59,11 +58,18 @@ cached-path = { version = "0.5", optional = true }
 aho-corasick = "0.7"
 paste = "1.0.6"
 proc_macros = { path = "./src/utils/proc_macros" }
+once_cell = "1.8"
```

Comment:
I don't think we need that either.

Even keeping the idea of `fancy-regex`, a simple enum is OK.
And even then, we **don't** want an enum. We want something that uses the correct regex engine ( `onig` or `onig-rs`) directly at compilation time.